src/main/java/mate/academy/spring/controller/MovieSessionController.java

```diff
@@ -0,0 +1,66 @@
+package mate.academy.spring.controller;
+
+import static java.util.stream.Collectors.toList;
+
+import java.time.LocalDate;
+import java.util.List;
+import mate.academy.spring.model.MovieSession;
+import mate.academy.spring.model.dto.MovieSessionRequestDto;
+import mate.academy.spring.model.dto.MovieSessionResponseDto;
+import mate.academy.spring.service.MovieSessionService;
+import mate.academy.spring.service.mapper.MovieSessionMapper;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.format.annotation.DateTimeFormat;
+import org.springframework.web.bind.annotation.DeleteMapping;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.PutMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.bind.annotation.RestController;
+
+@RestController
+@RequestMapping("/movie-sessions")
+public class MovieSessionController {
+    private final MovieSessionService movieSessionService;
+    private final MovieSessionMapper movieSessionMapper;
+
+    @Autowired
+    public MovieSessionController(MovieSessionService movieSessionService,
+                                  MovieSessionMapper movieSessionMapper) {
+        this.movieSessionService = movieSessionService;
+        this.movieSessionMapper = movieSessionMapper;
+    }
+
+    @PostMapping
+    public MovieSessionResponseDto save(
+            @RequestBody MovieSessionRequestDto movieSessionRequestDto) {
+        return movieSessionMapper.toDto(movieSessionService.add(
+                movieSessionMapper.toModel(movieSessionRequestDto)));
+    }
+
+    @GetMapping("/available")
+    public List<MovieSessionResponseDto> getAllAvailableByMovieIdAndDate(
+            @RequestParam String movieId, @RequestParam @DateTimeFormat(pattern = "yyyy-MM-dd")
+            LocalDate date) {
+        return movieSessionService.findAvailableSessions(Long.parseLong(movieId), date)
+                .stream()
+                .map(movieSessionMapper::toDto)
+                .collect(toList());
```

Comment:
```suggestion
            @RequestParam Long movieId, @RequestParam @DateTimeFormat(pattern = "yyyy-MM-dd")
            LocalDate date) {
        return movieSessionService.findAvailableSessions(movieId, date)
                .stream()
                .map(movieSessionMapper::toDto)
                .collect(toList());
```

You can just accept Long